test/test_compile_package.py

```diff
@@ -0,0 +1,374 @@
+# Owner(s): ["oncall: export"]
+
+import copy
+import glob
+import inspect
+import os
+import shutil
+import types
+import unittest
+
+import torch
+from torch._inductor.runtime.runtime_utils import cache_dir
+from torch._inductor.test_case import TestCase
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    requires_cuda,
+    run_tests,
+)
+
+
+class ClassForTest:
+    def __init__(self):
+        self.p = torch.tensor(10)
+
+    def g(self, a, b):
+        for i in range(30):
+            a = a + b * i + self.p
+        return a
+
+
+class TestCompilePackage(TestCase):
+    def path(self):
+        return os.path.join(cache_dir(), f"package_{self.id()}")
+
+    def _test_save(self, compiled_fn, args, expected, path, index) -> int:
+        # From a clean state, save_package() can produce some archive
+        # files and the files loaded back can reproduce the expected result.
+        # Returns the index of compilation hit in the cache.
+        load_dir = path
+        load_paths = glob.glob(os.path.join(load_dir, "*.dynamo_code"))
+        self.assertEqual(len(load_paths), 0)
+
+        compiled_fn.save_package()
```

Comment:
Why not take the path here?  Seems a bit weird to have the path be an arg to torch.compile (which won't do anything unless you call save).